routes/variableurl.go

```diff
@@ -0,0 +1,123 @@
+package routes
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type (
+	parseIndex uint
+
+	ArgName string
+
+	ParseInt struct {
+		ArgName ArgName
+	}
+	ParseFloat32 struct {
+		ArgName ArgName
+	}
+	ParseFloat64 struct {
+		ArgName ArgName
+	}
+	ParseString struct {
+		ArgName ArgName
+	}
+	ParseVoid struct {
+		Required string
+	}
+	ParseEnd struct{}
+)
+
+const (
+	index = parseIndex(iota)
+)
+
+var (
+	badURLTypeMessage  = ErrorBody{"Bad URL type."}
+	badURLTypeCode     = 400
+	notFoundURLMessage = ErrorBody{"Not Found."}
+	notFoundURLCode    = 404
+)
+
+func incrementIndex(a Arguments) int {
+	if _, ok := a[index]; ok {
+		a[index] = a[index].(int) + 1
+	} else {
+		a[index] = 1
+	}
+	return a[index].(int)
+}
+
+func (d ParseInt) Decorate(h IntermediateHandler) IntermediateHandler {
+	return func(w http.ResponseWriter, r *http.Request, a Arguments) (status int, output interface{}) {
+		splitURL := strings.Split(r.URL.Path, "/")
+		index := incrementIndex(a) - 1
+		if len(splitURL) > index {
+			if i, err := strconv.Atoi(splitURL[index]); err == nil {
+				a[d.ArgName] = i
+				return h(w, r, a)
+			}
+			return badURLTypeCode, badURLTypeMessage
+		}
+		return notFoundURLCode, notFoundURLMessage
+	}
+}
+
+func (d ParseFloat32) Decorate(h IntermediateHandler) IntermediateHandler {
```

Comment:
I don’t think defining this decorator for floats is necessary, also there is some complexity in handling floats which I think we should avoid : for example your parser allows "NaN", "+Inf", "-Inf", or "-0" which would be passed to a handler which may or may not be able to deal with those.
Unless you have an immediate example of why this is useful, I think you should drop `ParseFloat32` and  `ParseFloat64`.